src/wallet/rpcwallet.cpp

```diff
@@ -217,12 +215,8 @@ static void SetFeeEstimateMode(const CWallet* pwallet, CCoinControl& cc, const U
             throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Selected estimate_mode %s requires a fee rate to be specified in conf_target", estimate_mode.get_str()));
         }
 
-        CAmount fee_rate = AmountFromValue(estimate_param);
-        if (cc.m_fee_mode == FeeEstimateMode::SAT_B) {
-            fee_rate /= WALLET_BTC_KB_TO_SAT_B;
-        }
-
-        cc.m_feerate = CFeeRate(fee_rate);
+        CAmount feerate{AmountFromValue(estimate_param)};
+        cc.m_feerate = cc.m_fee_mode == FeeEstimateMode::SAT_B ? CFeeRate(feerate, COIN) : CFeeRate(feerate);
```

Comment:
Yes :+1: and I was more confident about making this change after all the test coverage we added in #20220.

Reply:
It would be nice to have a comment explaining this because it is not immediately clear why this works and future work on this code may not see this explanation.

Reply:
Done, added doxygen documentation to the CFeeRate ctor.

Reply:
Done, added doxygen documentation.

```diff
-    /** Constructor for a fee rate in satoshis per kB. The size in bytes must not exceed (2^63 - 1)*/
+
+    /** Constructor for a fee rate in satoshis per kvB (sat/kvB). The size in bytes must not exceed (2^63 - 1).
+     *
+     *  Passing an nBytes value of COIN (1e8) returns a fee rate in satoshis per vB (sat/vB),
+     *  e.g. (nFeePaid * 1e8 / 1e3) == (nFeePaid / 1e5),
+     *  where 1e5 is the ratio to convert from BTC/kvB to sat/vB.
+     *
+     *  @param[in] nFeePaid  CAmount fee rate to construct with
+     *  @param[in] nBytes    size_t units to construct with, if > 0
+     *  @returns   fee rate
+     */
     CFeeRate(const CAmount& nFeePaid, size_t nBytes);
```

Reply:
Done, added Doxygen documentation in a3eac6e603fddeda1dae20fba840d42f02710531

Reply:
I asked myself why else the two-argument ctor might have been written and didn't come up with an answer...I'm not against reviewing a refactoring of it.

Reply:
Looking at settxfee in `wallet/rpcwallet.cpp`, this method is used in a similar manner.
```
    CAmount nAmount = AmountFromValue(request.params[0]);
    CFeeRate tx_fee_rate(nAmount, 1000);
```

Reply:
> This feels like a hack, unless I misunderstand the original purpose of this constructor. Maybe it's better to add a separate constructor with a boolean to distinguish `SAT_VB` from `BTC/kvB`.

@Sjors done, proposed a separate constructor in https://github.com/bitcoin/bitcoin/commit/9c479bfc293fec9063acb9667ddb3cb60b46469d